### PR TITLE
PoolMetadataRegistry -> Check if the caller is the owner before set the PoolMetadata

### PR DESCRIPTION
**Pull request by devjoaov**
[linear issue](https://linear.app/bleu-llc/issue/BAL-85/check-if-the-caller-is-the-owner)

### CODE REVIEW

File: packages/contracts/test/PoolMetadataRegistry.t.sol [hunk 1]
```diff
@@ -71,20 +71,26 @@ contract PoolMetadataRegistryTest is PoolMetadataRegistryEvents, Test {
 
     function testIfUpdatePoolMetadataRevert() public {
         vm.expectRevert("Pool not registered");
-
         _poolMetadataRegistry.setPoolMetadata(0x0, _testMetadataCID);
```
Comment (ribeirojose) [inline packages/contracts/test/PoolMetadataRegistry.t.sol:74]: was this test passing before? 👀

File: packages/contracts/test/PoolMetadataRegistry.t.sol [hunk 2]
```diff
@@ -63,30 +68,43 @@ contract PoolMetadataRegistryTest is IPoolMetadataRegistry, Test {
...
+    function testIsPoolOwnerReturnsTrueWhenOwner() public {
         assertFalse(_poolMetadataRegistry.isPoolOwner(_basePool.getPoolId()));
+    }
 
-        vm.startPrank(address(15));
+    function testIsPoolOwnerFalseWhenNotOwner() public {
+        vm.startPrank(_poolOwner);
         assertTrue(_poolMetadataRegistry.isPoolOwner(_basePool.getPoolId()));
         vm.stopPrank();
     }
```
Comment (mendesfabio) [inline packages/contracts/test/PoolMetadataRegistry.t.sol:79]: these tests look inverted to me, no?

File: packages/contracts/test/PoolMetadataRegistry.t.sol [hunk 3]
```diff
@@ -63,30 +68,43 @@ contract PoolMetadataRegistryTest is IPoolMetadataRegistry, Test {
...
 
-    function testIfUpdatePoolMetadataRevert() public {
+    function testIfsetPoolMetadataRevertWhePoolNotRegistered() public {
         vm.expectRevert("Pool not registered");
 
         _poolMetadataRegistry.setPoolMetadata(0x0, _testMetadataCID);
     }
 
+    function testIfsetPoolMetadataRevertWhenNotOwner() public {
+        bytes32 notPoolId = _basePool.getPoolId();
```
Comment (mendesfabio) [inline packages/contracts/test/PoolMetadataRegistry.t.sol:88]: why is it called `notPoolId`?

File: packages/contracts/test/PoolMetadataRegistry.t.sol (hunk 2 above)
Comment (devjoaov) [reply packages/contracts/test/PoolMetadataRegistry.t.sol:79]: Yep, fixing it now

File: packages/contracts/test/PoolMetadataRegistry.t.sol (hunk 3 above)
Comment (devjoaov) [reply packages/contracts/test/PoolMetadataRegistry.t.sol:88]: Just a misunderstanding, fixing it in the next commit

**Review by ribeirojose — APPROVED** [review]

**Pull request merged**